### PR TITLE
fix: fixed style of go to top button

### LINKED ISSUE #3260: Footer button implementation issue
The `Go to top` button's body is not part of the button's active area. 

It is also not round, and it probably should be.

Image

### PR DESCRIPTION
**Pull request by berezinant**
fixes #3260

### After
image

### Before
image

### CODE REVIEW

**Review by IgnatBeresnev — APPROVED** [review]
I'm personally OK with the template/style changes, but let's merge it _after_ 1.9.20 code freeze (the end of this week). There will be breaking html/css changes in 2.0 anyway, so it makes sense to batch them

File: dokka-subprojects/plugin-base/src/main/resources/dokka/templates/includes/footer.ftl [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
-                href="https://github.com/Kotlin/dokka"><span>dokka</span><span class="padded-icon"></span></a></span>
-      </div>
+    <div class="footer">
+        <a href="#content" id="go-to-top-link" class="footer--button footer--button_go-to-top"></a>
+        <span>${footerMessage}</span>
+        <span class="pull-right">
+            <span>Generated by </span>
+            <a class="footer--link" href="https://github.com/Kotlin/dokka">
+                <span>dokka</span>
+                <span class="footer--external-link-icon"></span>
```
Comment (zoobestik) [inline dokka-subprojects/plugin-base/src/main/resources/dokka/templates/includes/footer.ftl:9]: If we change node for me `::after` looks better.

File: dokka-subprojects/plugin-base/src/main/resources/dokka/styles/style.css [hunk 2]
```diff
@@ -1432,46 +1445,51 @@ div.runnablesample {
...
-
-.footer > span {
-    padding: 0 16px;
+    margin-right: 8px;
 }
 
-.footer a {
+.footer--link {
+    display: inline-flex;
+    align-items: center; ;
```
Comment (zoobestik) [inline dokka-subprojects/plugin-base/src/main/resources/dokka/styles/style.css:1479]: `; ;`

**Review by zoobestik — APPROVED** [review]

File: dokka-subprojects/plugin-base/src/main/resources/dokka/templates/includes/footer.ftl (hunk 1 above)
Comment (berezinant) [reply dokka-subprojects/plugin-base/src/main/resources/dokka/templates/includes/footer.ftl]: Definitely! changed this

**Pull request merged**